fbpic/boundaries/moving_window.py

```diff
@@ -412,3 +544,39 @@ def shift_field_array_gpu( field_array, field_buffer, n_move ):
             # Set the remaining values to zero
             if (i+n_move) >= field_array.shape[0] or (i+n_move) < 0:
                 field_buffer[i, j] = 0.
+
+    @cuda.jit('void(complex128[:,:], complex128[:], int32)')
+    def shift_spect_array_gpu( field_array, shift_factor, n_move ):
+        """
+        Shift the field 'field_array' by n_move cells on the GPU.
+        This is done in spectral space and corresponds to multiplying the
+        fields with the factor exp(i*kz_true*dz)**n_move .
+
+        Parameters
+        ----------
+        field_array: 2darray of complexs
+            Contains the value of the fields, and is modified by
+            this function
+
+        shift_factor: 1darray of complexs
+            Contains the shift array, that is multiplied to the fields in
+            spectral space to shift them by one cell in spatial space
+            ( exp(i*kz_true*dz) )
+
+        n_move: int
+            The number of cells by which the grid should be shifted
+        """
```

Comment:
Could we actually use `iz` and `ir` here, instead of `i` and `j` ? I think that would be clearer.
(The same would apply to the above kernel.)